### ISSUE TITLE
Change of code semantics (bug)

### ISSUE THREAD

**Issue by fkling** [ForbesLindesay/halting-problem#2]
(sorry of the vague title)

The following code example contains two statements, a VariableDeclaration and an ExpressionStatement (containing a function call):

``` javascript
var foo = function() {}
bar()
```

The line break followed by the function call lead to automatic semicolon insertion.

However, `loopProtect` instruments the code like this:

``` javascript
var foo = function() {}
(haltingProblem.protect(2),bar())

// same as
var foo = function() {}(haltingProblem.protect(2),bar())
```

Now the function is executed immediately, instead of being assigned to `foo`.

One way to solve this would be to insert a `;` before the CallExpressions, if the parent node of the CallExpression is an ExpressionStatement. You could use `walk.ancestor` instead of `walk.simple` to get this information.

However, there might be a better way to solve the issue, e.g. simply ignoring ExpressionStatements that only contain a CallExpression. Since I'm not familiar with the reasons which expressions to instrument, I thought I'll leave this up to you.

**Comment by ForbesLindesay** [comment]
The problem this is attempting to solve is that newer JavaScript engines will start supporting tail-recursion, which allows for an infinite loop that looks like:

``` js
function foo() {
  return foo();
}
foo();
```

it wouldn't be a problem currently as most environments don't support tail-recursion but it may be an issue in the future.

**Comment by ForbesLindesay** [comment]
Fixed by 67b2a56bb765d7ed67cc2b2618ed3b0723df78fb

Sorry it took me so long to get round to this.
